.github/workflows/bk-ci.yml

```diff
@@ -0,0 +1,419 @@
+
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: mvn package
+        run: mvn -B -nsu clean package -DskipTests
+
+  macos-build:
+    name: Build with macos on JDK 11
+    runs-on: macos-latest
+    timeout-minutes: 30
+    needs: [ 'build-and-license-check' ]
+    if: ${{ needs.build-and-license-check.outputs.docs_only != 'true' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Tune Runner VM
+        uses: ./.github/actions/tune-runner-vm
+
+      - name: Cache local Maven repository
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/bookkeeper
+            !~/.m2/repository/org/apache/distributedlog
+          key: ${{ runner.os }}-bookkeeper-all-${{ hashFiles('**/pom.xml') }}
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: mvn package
+        run: mvn -B -nsu clean package -DskipTests
+
+  jdk-compatibility-checks:
+    name: ${{ matrix.step_name }}
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ matrix.timeout || 60 }}
+    needs: [ 'build-and-license-check' ]
+    if: ${{ needs.build-and-license-check.outputs.docs_only != 'true' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - step_name: Compatibility Check Java8
+            jdk_version: 8
+          - step_name: Compatibility Check Java11
+            jdk_version: 11
+          - step_name: Compatibility Check Java17
+            jdk_version: 17
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Tune Runner VM
+        uses: ./.github/actions/tune-runner-vm
+
+      - name: Cache local Maven repository
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/bookkeeper
+            !~/.m2/repository/org/apache/distributedlog
+          key: ${{ runner.os }}-bookkeeper-all-${{ hashFiles('**/pom.xml') }}
+
+
+      - name: Set up JDK ${{ matrix.jdk_version }}
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.jdk_version }}
+
+      - name: Build with Maven
+        run: mvn clean package -B -nsu -DskipBookKeeperServerTests -Dorg.slf4j.simpleLogger.defaultLogLevel=INFO
+
+      - name: print JVM thread dumps when cancelled
+        if: cancelled()
+        run: ./dev/ci-tool print_thread_dumps
+
+  owasp-dependency-check:
+    name: OWASP Dependency Check
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs: [ 'build-and-license-check' ]
+    if: ${{ needs.build-and-license-check.outputs.need_owasp == 'true' }}
```

Comment:
A question: Do we need to skip this job when the change belongs to `docs` in `.github/changes-filter.yaml` file ?